banzai/qc/header_checker.py

```diff
@@ -0,0 +1,155 @@
+"""
+This module performs controls on several critical parameters of image header, contains in
+header_expected_format dictionnary.
+@author:ebachelet
+"""
+from banzai.stages import Stage
+from banzai import logs
+
+class HeaderSanity(Stage):
+    """
+       This class contains the needed function/definitions to performs the header check.
+    """
+
+    def __init__(self, pipeline_context):
+        super(HeaderSanity, self).__init__(pipeline_context)
+
+        self.header_expected_format = {'RA': str, 'DEC': str, 'CAT-RA': str, 'CAT-DEC': str,
+                                       'OFST-RA': str, 'OFST-DEC': str, 'TPT-RA': str,
+                                       'TPT-DEC': str, 'PM-RA': str, 'PM-DEC': str,
+                                       'CRVAL1': float, 'CRVAL2': float, 'CRPIX1': int,
+                                       'CRPIX2': int, 'EXPTIME': float}
+
+    @property
+    def group_by_keywords(self):
+        return None
+
+    def do_stage(self, images):
+        """ Performs several checks on the header, see the relative functions.
+        :param list images: a list of image object.
+        :returns: the list of image object after check
+        :rtype: list
+        """
+        for image in images:
+
+
+            for keyword in self.header_expected_format.keys():
+                self.check_header_keyword_present(keyword, image)
+                self.check_header_format(keyword, image)
+                self.check_header_na(keyword, image)
+
+            self.check_ra_range(image)
+            self.check_dec_range(image)
+            self.check_exptime_value(image)
+
+        return images
+
+    def check_header_keyword_present(self, keyword, image):
+        """ Return a warning if the keyword is not in the image header.
+        :param str keyword: the stry keyword you look inside the header
+        :param object image: an image object
+        """
+        header = image.header
+
+        logging_tags = logs.image_config_to_tags(image, self.group_by_keywords)
+
+        if keyword not in header:
+            sentence = 'WARNING : The header key ' + keyword + ' is not in image header!'
```

Comment:
Don't have warning in the message itself. Use logger.warn and that will produce the warning message correctly.